File: src/main/java/cn/lnkdoc/sdk/uia/instance/bjtoon/request/UserInfoRequest.java
package cn.lnkdoc.sdk.uia.instance.bjtoon.request;

import cn.lnkdoc.sdk.uia.common.HttpMethod;
import cn.lnkdoc.sdk.uia.common.exception.UiaException;
import cn.lnkdoc.sdk.uia.common.property.IUiaProperty;
import cn.lnkdoc.sdk.uia.common.request.AbstractUiaRequest;
import cn.lnkdoc.sdk.uia.common.request.IUiaRequest;
import cn.lnkdoc.sdk.uia.common.util.Assert;
import cn.lnkdoc.sdk.uia.instance.bjtoon.property.BjtoonProperty;

/**
 * @author langkye
 * @since 1.0.0.RELEASE
 */
public class UserInfoRequest extends AbstractUiaRequest implements IUiaRequest {
    private String body;
    
    /**
     * request body
     *
     * @return request body
     */
    @SuppressWarnings("unchecked")
    @Override
    public String body() {
        return getBody();
    }

    /**
     * request url
     *
     * @param p property
     * @return request url
     */
    @Override
    public String url(IUiaProperty p) {
        BjtoonProperty property = (BjtoonProperty) p;
        Assert.required(property, "北京通配置不能为空");
        try {
            return
                    property.getDomain()
                            + property.getUserInfoPath()
                            + "?"
                            + "access_token=" + this.body
                    ;
        } catch (Exception e) {
            throw new UiaException(e);
        }
    }

    /**
     * request method
     *
     * @return request method
     */
    @Override
    public HttpMethod method() {
        return HttpMethod.GET;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }
}

File: src/main/kotlin/cn/lnkdoc/sdk/uia/instance/gitlab/request/UserInfoRequest.kt
package cn.lnkdoc.sdk.uia.instance.gitlab.request

import cn.lnkdoc.sdk.uia.common.HttpMethod
import cn.lnkdoc.sdk.uia.common.exception.UiaException
import cn.lnkdoc.sdk.uia.common.property.IUiaProperty
import cn.lnkdoc.sdk.uia.common.request.AbstractUiaRequest
import cn.lnkdoc.sdk.uia.common.request.IUiaRequest
import cn.lnkdoc.sdk.uia.common.util.Assert
import cn.lnkdoc.sdk.uia.instance.gitlab.property.GitlabProperty

/**
 * @author langkye
 * @since 1.0.0.RELEASE
 */
@Suppress("Unchecked_cast")
class UserInfoRequest: AbstractUiaRequest(), IUiaRequest {
    var body: String? = null
    
    /**
     * request body
     *
     * @param <T> type
     * @return request body
     */
    override fun <T> body(): T {
        return body as T
    }

    /**
     * request url
     *
     * @param property property
     * @return request url
     */
    override fun url(property: IUiaProperty): String {
        val varProperty = property as GitlabProperty
        Assert.required(varProperty, "配置不能为空")
        return try {
            (varProperty.userInfoUrl + "?access_token=${body}")
        } catch (e: Exception) {
            throw UiaException(e)
        }
    }

    /**
     * request method
     *
     * @return request method
     */
    override fun method(): HttpMethod {
        return HttpMethod.GET
    }
}